app/core/src/main/java/com/fsck/k9/contact/ContactIntentHelper.kt

```diff
@@ -0,0 +1,60 @@
+package com.fsck.k9.contact
+
+import android.content.Intent
+import android.net.Uri
+import android.provider.ContactsContract
+import com.fsck.k9.mail.Address
+
+object ContactIntentHelper {
+    @JvmStatic
+    fun getContactPickerIntent(): Intent {
+        return Intent(Intent.ACTION_PICK, ContactsContract.CommonDataKinds.Email.CONTENT_URI)
+    }
+
+    /**
+     * Get Intent to add information to an existing contact or add a new one.
+     *
+     * @param address An {@link Address} instance containing the email address
+     *              of the entity you want to add to the contacts. Optionally
+     *              the instance also contains the (display) name of that
+     *              entity.
+     */
+    fun getAddEmailContactIntent(address: Address): Intent {
+        val contactUri = Uri.fromParts("mailto", address.address, null)
+
+        val contactIntent = Intent(ContactsContract.Intents.SHOW_OR_CREATE_CONTACT)
+        contactIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+        contactIntent.data = contactUri
+        contactIntent.putExtra(
+            ContactsContract.Intents.EXTRA_CREATE_DESCRIPTION,
+            address.toString()
+        )
+
+        val name = address.personal
+        if (name != null) {
+            contactIntent.putExtra(
+                ContactsContract.Intents.Insert.NAME,
+                name
+            )
+        }
+
+        return contactIntent
```

Comment:
I like using `apply()` for cases like this. Makes it a bit easier to see at a glance that all of this code just builds a single `Intent`.